internal/api/impl/v1/dashboard/schemas/validator.go

```diff
@@ -125,35 +186,47 @@ func (v *validator) Validate(panels map[string]json.RawMessage) error {
 	return res
 }
 
-// LoadSchemas load the known list of schemas into the validator
-func (v *validator) LoadSchemas() {
-	pluginsPath := filepath.Join(v.schemasConf.Path, v.schemasConf.ChartsFolder)
+// LoadCharts loads the list of available charts plugins as CUE schemas
+func (v *validator) LoadCharts() {
+	v.charts.load()
+}
+
+// LoadQueries loads the list of available queries plugins as CUE schemas
+func (v *validator) LoadQueries() {
+	v.queries.load()
+}
+
+type cueDefs struct {
+	context     *cue.Context
+	baseDef     cue.Value
+	schemas     *sync.Map
+	schemasPath string
+	kindCuePath string
+}
 
-	files, err := os.ReadDir(pluginsPath)
+// load the list of available plugins as CUE schemas
+func (c *cueDefs) load() {
+	files, err := os.ReadDir(c.schemasPath)
 	if err != nil {
-		logrus.WithError(err).Errorf("Not able to read from schemas dir %s", pluginsPath)
+		logrus.WithError(err).Errorf("Not able to read from schemas dir %s", c.schemasPath)
 		return
 	}
 
-	// remove any previous item before proceeding
-	v.schemas.Range(func(key interface{}, value interface{}) bool {
-		v.schemas.Delete(key)
-		return true
-	})
+	// newSchemas is used for double buffering, to avoid any issue when there are panels to validate at the same time load() is triggered
+	newSchemas := sync.Map{}
```

Comment:
here it can be a simple map not necessary a `sync.Map` since this method is thread safe.